crates/router/src/db/business_profile.rs

```diff
@@ -114,38 +114,118 @@ impl BusinessProfileInterface for Store {
+        };
+
+        business_profiles.push(business_profile_insert.clone());
+
+        Ok(business_profile_insert)
     }
 
     async fn find_business_profile_by_profile_id(
         &self,
-        _profile_id: &str,
+        profile_id: &str,
     ) -> CustomResult<business_profile::BusinessProfile, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        match self
+            .business_profiles
+            .lock()
+            .await
+            .iter()
+            .find(|business_profile| business_profile.profile_id == profile_id)
+        {
+            Some(business_profile) => Ok(business_profile.clone()),
+            None => {
+                return Err(errors::StorageError::ValueNotFound(
+                    "business profile not found".to_string(),
+                )
+                .into())
+            }
+        }
     }
 
     async fn update_business_profile_by_profile_id(
         &self,
-        _current_state: business_profile::BusinessProfile,
-        _business_profile_update: business_profile::BusinessProfileUpdateInternal,
+        current_state: business_profile::BusinessProfile,
+        business_profile_update: business_profile::BusinessProfileUpdateInternal,
     ) -> CustomResult<business_profile::BusinessProfile, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        match self
+            .business_profiles
+            .lock()
+            .await
+            .iter_mut()
+            .find(|bp| bp.profile_id == current_state.profile_id)
+            .map(|bp| {
+                let business_profile_updated =
+                    business_profile_update.apply_changeset(current_state);
+                *bp = business_profile_updated.clone();
+                business_profile_updated
+            }) {
+            Some(business_profile) => Ok(business_profile),
+            None => {
+                return Err(errors::StorageError::ValueNotFound(
+                    "cannot find business profile to update".to_string(),
+                )
+                .into())
+            }
+        }
     }
 
     async fn delete_business_profile_by_profile_id_merchant_id(
         &self,
-        _profile_id: &str,
-        _merchant_id: &str,
+        profile_id: &str,
+        merchant_id: &str,
     ) -> CustomResult<bool, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        let mut business_profiles = self.business_profiles.lock().await;
+
+        match business_profiles
+            .iter()
+            .position(|bp| bp.profile_id == profile_id && bp.merchant_id == merchant_id)
+        {
+            Some(index) => {
+                _ = business_profiles.remove(index);
+                Ok(true)
+            }
+            None => Err(errors::StorageError::ValueNotFound(
+                "cannot find business profile to delete".to_string(),
+            )
+            .into()),
+        }
```

Comment:
Same here, you can improve this by changing this into like this:

```suggestion
        let index = business_profiles
            .iter()
            .position(|bp| bp.profile_id == profile_id && bp.merchant_id == merchant_id)
            .ok_or(errors::StorageError::ValueNotFound(
                    "cannot find business profile to delete".to_string(),
                )
                .into()
            )?;
            business_profiles.remove(index);
            Ok(true)
```